bug_actiongroup.php

```diff
@@ -202,20 +204,26 @@
 		case 'UP_STATUS':
 			$f_status = gpc_get_int( 'status' );
 			if( access_has_bug_level( access_get_status_threshold( $f_status, $t_bug->project_id ), $t_bug_id ) ) {
-				if( true == bug_check_workflow( $t_status, $f_status ) ) {
+				if( true == bug_check_workflow( $t_status, $f_status )
+					|| access_compare_level( current_user_get_access_level(), config_get_access( 'status_enum_workflow' ) )
+					) {
 					# @todo we need to issue a helper_call_custom_function( 'issue_update_validate', array( $t_bug_id, $t_bug_data, $f_bugnote_text ) );
-					bug_set_field( $t_bug_id, 'status', $f_status );
-
-					# Add bugnote if supplied
+					$t_status_changed = $t_status !== $f_status;
 					if( !is_blank( $f_bug_notetext ) ) {
-						$t_bugnote_id = bugnote_add( $t_bug_id, $f_bug_notetext, null, $f_bug_noteprivate );
+						$t_bugnote_id = bugnote_add( $t_bug_id, $f_bug_notetext, null, $f_bug_noteprivate, BUGNOTE, '', null, !$t_status_changed );
 						bugnote_process_mentions( $t_bug_id, $t_bugnote_id, $f_bug_notetext );
-						# No need to call email_generic(), bugnote_add() does it
+					}
+					if( $t_status_changed ) {
+						bug_set_field( $t_bug_id, 'status', $f_status );
+						$t_new_status_label = MantisEnum::getLabel( config_get( 'status_enum_string' ), $f_status );
+						$t_new_status_label = str_replace( ' ', '_', $t_new_status_label );
+						email_bug_status_changed( $t_bug_id, $t_new_status_label );
 					} else {
-						email_bug_updated( $t_bug_id );
+						$t_failed_ids[$t_bug_id] = lang_get( 'bug_actiongroup_same_status' );
+					}
+					if( $t_status_changed || !is_blank( $f_bug_notetext ) ) {
+						helper_call_custom_function( 'issue_update_notify', array( $t_bug_id ) );
 					}
-
-					helper_call_custom_function( 'issue_update_notify', array( $t_bug_id ) );
```

Comment:
Why did you remove the issue_update_notify?  We used to trigger this before and trigger this in other events in the same group action file (e.g. update category).